pkg/cmd/server/origin/auth.go

```diff
@@ -152,12 +154,23 @@ func (c *AuthConfig) InstallAPI(container *restful.Container) []string {
 	osOAuthClientConfig.RedirectUrl = c.Options.MasterPublicURL + path.Join(OpenShiftOAuthAPIPrefix, tokenrequest.DisplayTokenEndpoint)
 
 	osOAuthClient, _ := osincli.NewClient(osOAuthClientConfig)
-	if c.MasterRoots != nil {
-		// Copy the default transport
-		var transport http.Transport = *http.DefaultTransport.(*http.Transport)
-		// Set TLS CA roots
-		transport.TLSClientConfig = &tls.Config{RootCAs: c.MasterRoots}
-		osOAuthClient.Transport = &transport
+	if c.Options.MasterCA != nil && len(*c.Options.MasterCA) > 0 {
+		rootCAs, err := cmdutil.CertPoolFromFile(*c.Options.MasterCA)
+		if err != nil {
+			glog.Fatal(err)
+		}
+
+		osOAuthClient.Transport = &http.Transport{
```

Comment:
> this is identical to the default transport... I'm pretty sure it was improper to create a copy of it and override it previously...

Definitely needs a comment indicating that the intent is to reproduce the defaults.  It's not obvious.